src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCompute.java

```diff
@@ -114,6 +122,10 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
 
         } catch (CompilationFailureException e) {
           throw new IllegalArgumentException(e.getMessage(), e);
+        } catch (IOException e) {
+          Throwables.propagate(e);
```

Comment:
use `throw new IllegalArgumentException(e);` instead.

Reply:
Looks like the `IOException` has nothing to do with the argument passed, should it still be thrown as `IllegalArgumentException` ?